### ISSUE TITLE
Does this editor actually work for people?

### ISSUE THREAD

**Issue by EvanMachusak** [Microsoft/vscode#32640]
This is probably a bit of a rant, but I am having widespread systemic issues trying to use Code for real - as in for professional development.

Basic usability features simply don't work - or work until they stop working, at which point an F1 -> Reload Window is required.

For example, TypeScript intellisense.  It works, then stops working.  Reload required.  This probably the root cause for why the F12 key stops working. 

Occasionally, the build command simply stops working.  Ctrl-shift-B stops doing anything.  Great.

Even though I have one task defined in tasks.json and it's set to be the default build command, when I press it, I still get a dropdown asking me to choose tsc as being the build command.  Why?!

A simple TSC->Webpack (both in watch mode) configuration appears to be basically impossible with the tasks.json unless I start using commands, gulp, or some other wizardry that I don't have the time to learn.  It really feels like whoever designed tasks.json's schema has never used an IDE before.  Why can't I just run two commands one after the other when I press ctrl-shift-B (and why isn't the default F5?)

And then of course there's the no multi-monitor support.  I know it's on the backlog, but it's a pretty big issue.

Finally - what's the deal with the 15+ Code.exe processes that are all using 100+mb of ram?  I feel like this increases every time "Reload window" is used, which is often.  Maybe my extensions are to blame.  Of course without some of them (e.g., TFVC) I can't really do my job so that's not a consolation.

I'm still using it because VS2017's quality has spiraled so far downward that it's even less usable than Code is right now with all its bugs for front-end dev with modern frameworks.

This editor has potential but these issues are holding it back, big time.  Please spend some more effort on the basics before cramming in more useless features like a CSS color palette picker.

**Comment by Adondriel** [comment]
if you're complaining about a bunch of 100+MB ram processes, you have bigger issues. But yea, this editor DOES work for people, and as this is a question, it should have been asked on the Stack Overflow.

**Comment by Adondriel** [comment]
Could you comment as to what extensions you are using? and/or what type of project you are working on.

**Comment by EvanMachusak** [comment]
Adam: I'm pretty sure I answered those questions in the original thread.  Also, we can do without the patronizing tone.  I am posting my concerns here because this is where development of Visual Studio Code happens and I am reporting errors that I cannot pinpoint to specific incidents or actions that I take in the editor.  Something like the F12 key should not simply stop working, but it does.  All the time.

I've disabled every addon except TFVC and I'll see if that has any effect and update this issue accordingly.

**Comment by Adondriel** [comment]
the ticket you submitted has text(that you erased), that says "If you are asking a question, ask it on stack overflow" It's not me being patronizing, it's me attempting to help the devs of this project keep the 4,805 tickets slightly more on topic. The different processes are to keep different tasks on different threads, so that one task won't slow down the UI (ex: searching a large project, does this, cause it seems to run on the main thread... not sure if fixed yet.)

**Comment by EvanMachusak** [comment]
It's called a rhetorical question, Adam.

**Comment by Adondriel** [comment]
You're on Github, you ask a question, you get a technical answer. Rhetoric does not travel very well through text, without adding a "/s".
I'll be nice and help answer your question here: 
1. Typescript Intellisense, there are some defaults given with VSCode, but not many. I had to get the extension "Angular Language Service" in order to get them to work consistently. 
2. I don't use more than 1 build task, personally so I can't help much here, but VSCode has A LOT of support for actually launching and hosting your project. You should check out the launch.json config docs for more on that. https://code.visualstudio.com/docs/editor/debugging
3. Have you set your task up correctly? It looks as if it has params to allow you to set the file? I'm not sure, I've not messed with this stuff. https://code.visualstudio.com/docs/editor/tasks
4. This is supported, in Launch, where you can launch things, one after the other. This is assuming that your project has both a server and a client side, and isn't just an angular project. also, I think build does support running 2 cmds at once? I'm not sure, I glanced at the docs just briefly. 
5. You just need a bigger monitor. [This](http://www.dell.com/en-us/shop/dell-ultrasharp-34-curved-ultrawide-monitor-u3415w/apd/210-adtr/parts-upgrades) is my favorite monitor for coding.
6. This is due in part to 2 things, which help each other out. 1. It has to do with how Electron does stuff, which is what Code was written with, and 2. it helps to make sure the main thread doesn't get slowed down by extensions, searches, etc. It also has a side effect (I would infer, not sure if true or not) of making each extension sandboxed, so they can't screw with each other.

Yes, it's not perfect, and of course there are improvements. But there are also A LOT of settings you should try messing around with. It's not ever going to be possible to make an editor that fits everyone PERFECTLY because everyone is different. That's why we have the loads of different settings that we do.
Lots of detail on what different settings can do, on this page. https://code.visualstudio.com/docs/getstarted/userinterface

**Comment by aeschli** [comment]
Thanks @EvanMachusak for the experience report. Thanks @Adondriel for helping to answer.
Yes, StackOverflow is the right place for the 'how to setup' questions.

For issues like TypeScript server crashing, or build command not executing, Github is the right place. But we need a separate issue per problem, in order to work and discuss this.
Also, without steps to reproduce there's not much we can do.

Closing this, but it would be great if you could separate the various issues and enhancement requests, file separate issues that contain more information on your setup.

**Comment by aeschli** [comment]
@dbaeumer see the feedback on tasks.

**Comment by dbaeumer** [comment]
@aeschli thanks. The task 2.0 support offers to define a composite tasks consisting out of two tasks and run them either sequentially or in parallel. The property to use is `dependsOn`
